### DISCUSSION TITLE
Lidarr within Synology Docker container.

### DISCUSSION THREAD

**Discussion by KINGLIFER** [RandomNinjaAtk/arr-scripts#57] [Q&A] [answered]
For some reason this will install and run yet will NOT install the scripts. I found no other documentation besides to place the two files in folders. 

There is nothing to show since it doesn't work. There is not error besides this..
══════════════════════════════════════════════════════════════════╗
║                                                                         ║
║        Some of the contents of the folder /custom-cont-init.d           ║
║            are not owned by root, which is a security risk.             ║
║                                                                         ║
║  Please review the permissions of this folder and its contents to make  ║
║     sure they are owned by root, and can only be modified by root.      ║
║                                                                         ║
╚═════════════════════════════════════════════════════════════════════════╝

my compose is this.

version: "3.7"

services:
  lidarr:
    container_name: lidarr
    image: lscr.io/linuxserver/lidarr:latest
    ports:
      - 8686:8686
    environment:
      - PUID=0
      - PGID=0

      - TZ=America/NewYork
    volumes:
      - /volume1/docker/lidarr/config:/config
      - /volume1/docker/lidarr/folder-01:/custom-services.d
      - /volume1/docker/lidarr/folder-02:/custom-cont-init.d
      - /volume1/docker/lidarr/data/music:/music #optional
      - /volume1/docker/lidarr/data/downloads:/downloads #optional
    restart: unless-stopped

any help would be appreciated.

**Comment by RandomNinjaAtk** [comment]
Did you check the contents of the downloaded script to make sure it matched the file on GitHub?

Commonly, users are downloading the wrong thing. If the text in the file doesn’t match the one on here on GitHub, that is your issue.

edit: 
Example: https://github.com/RandomNinjaAtk/arr-scripts/issues/30

**Comment by KINGLIFER** [comment]
This is what I downloaded. 

#!/usr/bin/with-contenv bash
curl https://raw.githubusercontent.com/RandomNinjaAtk/arr-scripts/main/lidarr/setup.bash | bash
exit

**Reply by RandomNinjaAtk** [reply]
Try setting the permissions on the file (chmod 777) and restart the container… maybe for some reason however you downloaded it, the container may not be able to read the file?

I’m  just making a guess, if you can provide more of the log output from container start, that will help because you only show the first few lines, and not the entire startup…

If you know how to access the container cli, I could possibly give you a curl command to make it simpler…

**Reply by KINGLIFER** [reply]
ok will try that.

**Comment by KINGLIFER** [comment]
Results.
╔═════════════════════════════════════════════════════════════════════════╗
║                                                                         ║
║        Some of the contents of the folder /custom-services.d            ║
║      have write permissions for others, which is a security risk.       ║
║                                                                         ║
║  Please review the permissions of this folder and its contents to make  ║
║     sure they are owned by root, and can only be modified by root.      ║
║                                                                         ║
╚═════════════════════════════════════════════════════════════════════════╝
╔═════════════════════════════════════════════════════════════════════════╗
║                                                                         ║
║        Some of the contents of the folder /custom-cont-init.d           ║
║      have write permissions for others, which is a security risk.       ║
║                                                                         ║
║  Please review the permissions of this folder and its contents to make  ║
║     sure they are owned by root, and can only be modified by root.      ║
║                                                                         ║
╚═════════════════════════════════════════════════════════════════════════╝
[custom-init] Service files found in /custom-services.d
[custom-init] custom-services.d: is not a file
[migrations] started
[migrations] no migrations found
usermod: no changes
───────────────────────────────────────
      ██╗     ███████╗██╗ ██████╗ 
      ██║     ██╔════╝██║██╔═══██╗
      ██║     ███████╗██║██║   ██║
      ██║     ╚════██║██║██║   ██║
      ███████╗███████║██║╚██████╔╝
      ╚══════╝╚══════╝╚═╝ ╚═════╝ 
   Brought to you by linuxserver.io
───────────────────────────────────────
To support the app dev(s) visit:
Lidarr: https://opencollective.com/lidarr
To support LSIO projects visit:
https://www.linuxserver.io/donate/
───────────────────────────────────────
GID/UID
───────────────────────────────────────
User UID:    911
User GID:    911
───────────────────────────────────────
[custom-init] Files found, executing
[custom-init] custom-cont-init.d: is not a file
[Info] Bootstrap: Starting Lidarr - /app/lidarr/bin/Lidarr - Version 1.2.6.3313 
[Info] AppFolderInfo: Data directory is being overridden to [/config] 
[Debug] Bootstrap: Console selected 
[Info] AppFolderInfo: Data directory is being overridden to [/config] 
[Info] AppFolderInfo: Data directory is being overridden to [/config] 
[Info] MigrationController: *** Migrating data source=/config/lidarr.db;cache size=-10000;datetimekind=Utc;journal mode=Wal;pooling=True;version=3 *** 
[Info] MigrationController: *** Migrating data source=/config/logs.db;cache size=-10000;datetimekind=Utc;journal mode=Wal;pooling=True;version=3 *** 
[Info] Microsoft.Hosting.Lifetime: Now listening on: http://[::]:8686 
[ls.io-init] done.
[Info] Microsoft.Hosting.Lifetime: Application started. Press Ctrl+C to shut down. 
[Info] Microsoft.Hosting.Lifetime: Hosting environment: Production 
[Info] Microsoft.Hosting.Lifetime: Content root path: /app/lidarr/bin 
[Info] RssSyncService: Starting RSS Sync 
[Warn] FetchAndParseRssService: No available indexers. check your configuration. 
[Info] DownloadDecisionMaker: No results found 
[Info] RssSyncService: RSS Sync Completed. Reports found: 0, Reports grabbed: 0 
[Info] ImportListSyncService: Starting Import List Sync 
[Info] ImportListSyncService: Processing 0 list items 
[Info] ImportListSyncService: Import List Sync Completed. Items found: 0, Artists added: 0, Albums added: 0 






If I ssh and change the folder to 777 it just goes back to the same state.

**Reply by RandomNinjaAtk** [reply]
Try setting the folders and contents to 777

```
chmod 777 -R /volume1/docker/lidarr/folder-01
chmod 777 -R /volume1/docker/lidarr/folder-02
```

You can also try removing/deleting both folders, and then from the container CLI, run the following command:

`curl https://raw.githubusercontent.com/RandomNinjaAtk/arr-scripts/main/lidarr/scripts_init.bash -o /custom-cont-init.d/scripts_init.bash`

The folders should get re-created by docker when the container starts...

**Reply by KINGLIFER** [reply]
0Warning: Failed to open the file /custom-cont-init.d/scripts_init.bash: No
Warning: such file or directory
100   130  100   130    0     0    651      0 --:--:-- --:--:-- --:--:--   653
curl: (23) Failure writing output to destination

**Reply by RandomNinjaAtk** [reply]
That's at least some progress. Seems like your just having general file system issues, like not being able to write to a directory and etc. 

I don't have a synology to replicate and clearly others are using the scripts just fine. Must be something wrong with synology docker system or something else specific to your configuration. Maybe if someone else has experience with that system, the can chime in to help....

What your trying to do is this: https://docs.linuxserver.io/general/container-customization#custom-scripts

**Reply by KINGLIFER** [reply]
Oh... ok I see my config is not inside my docker (Appfolder if that makes sense) it is in a folder outside of that. I get why the folder is not adding. I am outside of root. Could that help in anyway? how to make your scripts install to /volume1/docker/lidarr/  or a way I can just download the packages myself and place them in folders?

**Comment by RandomNinjaAtk** [comment, marked as answer]
PORTAINER was causing the problem, it was resolved by using standard docker configurations on synology....

Lesson to be learned: DON'T USE PORTAINER!

Added warning: https://github.com/RandomNinjaAtk/arr-scripts#warning

More context: https://wiki.servarr.com/docker-guide#portainer

**Comment by KINGLIFER** [comment]
With help from PearsonFlyer and be6np68es over on Discord and special thanks to Sage2050 and RandomNinjaAtk who litterally took time to assist I can say the main issue was using Portainer

**Reply by RandomNinjaAtk** [reply]
https://wiki.servarr.com/docker-guide#portainer
